accounts/abi/abi_test.go

```diff
@@ -458,3 +458,142 @@ func TestBareEvents(t *testing.T) {
 		t.Error("expected 'name' event to be present")
 	}
 }
+
+func TestStringReturn(t *testing.T) {
+	const definition = `[
+	{ "name" : "multi", "const" : true, "outputs": [ { "name": "a", "type": "string" } ] }]`
+
+	abi, err := JSON(strings.NewReader(definition))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// using buff to make the code readable
+	buff := new(bytes.Buffer)
+	buff.Write(common.Hex2Bytes("0000000000000000000000000000000000000000000000000000000000000020"))
+	buff.Write(common.Hex2Bytes("0000000000000000000000000000000000000000000000000000000000000005"))
+	stringOut := "hello"
+	buff.Write(common.RightPadBytes([]byte(stringOut), 32))
+
+	out := abi.parseInterface("multi", buff.Bytes()).(string)
+	if out != stringOut {
+		t.Error("expected", stringOut, "got", out)
+	}
+}
+
+func TestBytesReturn(t *testing.T) {
+	const definition = `[
+	{ "name" : "multi", "const" : true, "outputs": [ { "name": "a", "type": "bytes" } ] }]`
+
+	abi, err := JSON(strings.NewReader(definition))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// using buff to make the code readable
+	buff := new(bytes.Buffer)
+	buff.Write(common.Hex2Bytes("0000000000000000000000000000000000000000000000000000000000000020"))
+	buff.Write(common.Hex2Bytes("0000000000000000000000000000000000000000000000000000000000000020"))
```

Comment:
Could we do a length of 5 here too? It would ensure that bytes are correctly parsed too and the padding discarded. That of course would modify `bytesOut` to contain only `[]byte("hello")`.